src/Kestrel.Core/Internal/Http2/Http2FrameWriter.cs

```diff
@@ -193,6 +193,48 @@ public void WriteResponseHeaders(int streamId, int statusCode, IHeaderDictionary
             }
         }
 
+        public Task WriteResponseTrailers(int streamId, HttpResponseTrailers headers)
+        {
+            lock (_writeLock)
+            {
+                if (_completed)
+                {
+                    return Task.CompletedTask;
+                }
+
+                _outgoingFrame.PrepareHeaders(Http2HeadersFrameFlags.END_STREAM, streamId);
+                var buffer = _headerEncodingBuffer.AsSpan();
+                var done = _hpackEncoder.BeginEncode(EnumerateHeaders(headers), buffer, out var payloadLength);
```

Comment:
What happens if a trailer is larger than the max response frame size?

Reply:
It will throw that HPackEncodingException I added, same as response headers.